src/v/kafka/server/handlers/describe_log_dirs.cc

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "kafka/server/handlers/describe_log_dirs.h"
+
+#include "cluster/partition_manager.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/server/request_context.h"
+#include "kafka/server/response.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/util/log.hh>
+
+#include <fmt/ostream.h>
+
+namespace kafka {
+
+using part_set
+  = absl::flat_hash_map<model::topic, std::vector<describe_log_dirs_partition>>;
+
+static describe_log_dirs_partition describe_partition(cluster::partition& p) {
+    return describe_log_dirs_partition{
+      .partition_index = p.ntp().tp.partition(),
+      .partition_size = static_cast<int64_t>(p.size_bytes()),
+      .offset_lag = 0,
+      .is_future_key = false,
+    };
+}
+
+static part_set collect_mapper(
+  cluster::partition_manager& pm,
+  const std::optional<std::vector<describable_log_dir_topic>>& topics) {
+    part_set ret;
+
+    /*
+     * return all partitions
+     */
+    if (!topics) {
+        for (auto& partition : pm.partitions()) {
+            if (partition.first.ns != model::kafka_namespace) {
+                continue;
+            }
+            ret[partition.first.tp.topic].push_back(
+              describe_partition(*partition.second));
+        }
+        return ret;
+    }
+
+    /*
+     * return only partition matching request
+     */
+    for (auto& topic : *topics) {
```

Comment:
Const?

Reply:
`topics` is already a const container

Reply:
Makes sense.

Reply:
mm, I like the explicit const even on the use cases in case the containers change.

Reply:
fixed